ghcjs-dom-jsffi/src/GHCJS/DOM/EventTargetClosures.hs

```diff
@@ -11,8 +19,18 @@ import GHCJS.Marshal.Pure
 import GHCJS.DOM.Types
 import GHCJS.Foreign.Callback.Internal
 
+-- | Plain JS 'DOMString' that carries two phantom types:
+--
+--   * @t@: type of the event’s target
+--   * @e@: type of the event
+--
+--   Many @GHCJS.DOM@ modules export @EventName@s, for example
+--   "GHCJS.DOM.JSFFI.Generated.Window".
 newtype EventName t e = EventName DOMString
+
+-- | Plain JS 'EventListener' that carries the same phantom types as 'EventName'.
 newtype SaferEventListener t e = SaferEventListener EventListener
+-- TODO: why is it prefixed with “Safer”, while EventName above just aliases?
```

Comment:
`SaferEventListener` is to `EventListener` what `EventName` is to `DOMString`.  If `SaferEventListener` did not have a prefix we would have two types called `EventListener`.  I was worried that might lead to confusion.